modules/api/php/endpoints/candidate/visit/dicoms.class.inc

```diff
@@ -138,20 +156,167 @@ class Dicoms extends Endpoint implements \LORIS\Middleware\ETagCalculator
+                'The request should have a "json" form attribute containing the '
+                    . 'request data.'
+            );
+        }
+
+        $data = json_decode($request_body['json'], true);
+
+        $input_candid = $data['CandID'] ?? null;
+        $visit_candid = $this->_visit->getCandID();
+        if ((string) $input_candid !== (string) $visit_candid) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'CandID in POST data do not match URL.'
+            );
+        }
+
+        $pscid = $data['PSCID'] ?? null;
+        if (!\Candidate::candidateExists($visit_candid, $pscid)) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'PSCID and CandID in POST data do not refer to the same candidate.'
+            );
+        }
+
+        $visit_label = $data['Visit'] ?? null;
+        if ($visit_label !== $this->_visit->getVisitLabel()) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'Visit label in URL and POST data do not match.'
+            );
+        }
+
+        $is_phantom = $data['IsPhantom'] ?? null;
+        if (!in_array($is_phantom, [true, false])) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'IsPhantom must be either `true` or `false`.'
+            );
+        }
+
+        $file_name    = $uploaded_file->getClientFilename();
+        $patient_name = $pscid . "_" . $input_candid . "_" . $visit_label;
+        if (!preg_match("/^{$patient_name}/", $file_name) && !$is_phantom) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                "Filename must start with $patient_name."
+            );
+        }
+
+        // *************************
+        // *** Upload file *********
+        // *************************
+        $incoming_path = \NDB_factory::singleton()
+            ->config()
+            ->getSetting('MRIUploadIncomingPath');
+
+        $incoming_dir = new \SplFileInfo($incoming_path);
+
+        $uploader = (new \LORIS\FilesUploadHandler($incoming_dir))
+            ->withPermittedMIMETypes(
+                'application/gzip',
+                'application/tar',
+                'application/tar+gzip',
+                'application/x-tar'
+            );
+
+        if ($request->getHeaderLine('LORIS-Overwrite') === 'overwrite') {
+            $uploader = $uploader->withOverwrite(true);
+        }
+
+        $upload_response = $uploader->handle($request);
+        if ($upload_response->getStatusCode() !== 201) {
+            return $upload_response;
+        }
+
+        // *************************
+        // *** Update Database *****
+        // *************************
+        $file_path = $incoming_dir->getPathname() . '/' . $file_name;
+        $file_info = new \SplFileInfo($file_path);
+        if (!$file_info->isFile()) {
+            return new \LORIS\Http\Response\JSON\InternalServerError(
+                'Upload failed (see error_log.)'
```

Comment:
what `error_log` is this referring to? The apache error log? If so, most users do not have access to that file so I would probably rephrase that to contact your LORIS admin to check the error_log with a timestamp to ease parsing.

Reply:
I must say I am not sure which error log this is. It was part of the original PR. I removed it in favor of asking to contact the LORIS administrator. This certainly could benefit from more information but I did not find any good example to draw aspiration from in the codebase, so I stayed pretty basic.